### PR TITLE
chore: add client mapping info to glossary

### PR DESCRIPTION
**Pull request by justinabrahms**
Refs: https://github.com/open-feature/java-sdk/pull/388

### CODE REVIEW

**Review by toddbaert — APPROVED** [review]

File: specification/glossary.md [hunk 1]
```diff
@@ -92,7 +92,9 @@ A source-of-truth for flag values and rules. Flag management systems may include
 
 ### Provider
 
-An SDK-compliant implementation which resolves flag values from a particular flag management system, allowing the use of the [Evaluation API](./sections/01-flag-evaluation.md#13-flag-evaluation) as an abstraction for the system in question.
+An SDK-compliant implementation which resolves flag values from a particular flag management system, allowing the use of the [Evaluation API](./sections/01-flag-evaluation.md#13-flag-evaluation) as an abstraction for the system in question. 
+
+Providers can be used in two ways. Client-specific providers are active for specific clients, based on their name. Default providers are used if there are no client-specific mappings setup.
```
Comment (toddbaert) [inline specification/glossary.md:97]:
```suggestion
Providers can be used in two ways. Client-specific providers are active for specific clients, based on their name. The default provider is used if there are no client-specific mappings setup.
```

Is it better to use singular for this? I'm fine either way.

**Review by toddbaert — APPROVED** [review]
I don't think we need to wait for the full 5 approvals on this one. Once the CI finishes I'm fine with you force-merging, since this is non-normative and explanatory stuff.

**Review by beeme1mr — APPROVED** [review]

File: specification/glossary.md [hunk 2]
```diff
@@ -92,7 +92,9 @@ A source-of-truth for flag values and rules. Flag management systems may include
 
 ### Provider
 
-An SDK-compliant implementation which resolves flag values from a particular flag management system, allowing the use of the [Evaluation API](./sections/01-flag-evaluation.md#13-flag-evaluation) as an abstraction for the system in question.
+An SDK-compliant implementation which resolves flag values from a particular flag management system, allowing the use of the [Evaluation API](./sections/01-flag-evaluation.md#13-flag-evaluation) as an abstraction for the system in question.
```
Comment (beeme1mr) [inline specification/glossary.md:95]:
```suggestion
An SDK-compliant implementation which resolves flag values from a particular flag management system, allowing the use of the [Evaluation API](./sections/01-flag-evaluation.md#13-flag-evaluation) as an abstraction for the system in question.
```

**Comment by toddbaert** [comment]
Force merging this (editorial change).

**Pull request merged**